CMakeLists.txt

```diff
@@ -61,11 +61,22 @@ set(TOOLS ${CMAKE_SOURCE_DIR}/tools)
 # ==================================================================================================
 # Compiler check
 # ==================================================================================================
-if ("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-    message(FATAL_ERROR
-            "The GCC compiler is not supported, please use clang for both C and C++. "
-            "Please refer to README.md for more information.")
-    return()
+set(MIN_CLANG_VERSION "5.0")
+
+if (CMAKE_C_COMPILER_ID STREQUAL "Clang")
```

Comment:
`clang` on macOS is called `AppleClang`, you need to check for that compiler ID as well.